src/app/components/Ionic/IonicProvider.tsx

```diff
@@ -21,10 +26,25 @@ const IonicContextProvider: FC<PropsWithChildren> = ({ children }) => {
       }
     })
 
+    // TODO: appStateChange triggers on the web platform as well, using visibilitychange listener, consider reusing the code(downside @capacitor/app dependency on web & extension)
+    const appStateChangeListenerHandle = App.addListener('appStateChange', ({ isActive }) => {
```

Comment:
Well the interface of addListener is `Promise<PluginListenerHandle> & PluginListenerHandle`, so I would expect it to work either way.

Reply:
You can call `remove` method only when promise is resolved so we deal with a promise here, right?

Reply:
Yes, and based on this code:

https://github.com/ionic-team/capacitor/blob/f8264ccae1f2fec553521bc62d767c4909ea6d79/core/src/web-plugin.ts#L29-L59

It is resolved immediately. So no difference.

Reply:
If you say "resolved" this mean you deal with async code and current code does not handle promise.
Moreover, code is not future proof. It won't work in v6 which we will use anyway. If you look at new return type 
https://capacitorjs.com/docs/apis/app#addlistenerappstatechange-
This will just throw ts error.

Reply:
Ok, will improve so it's future proof.